src/modules/generateRandomNumber.js

```diff
@@ -1,15 +1,19 @@
-'use strict';
-
-/**
- * Generate a random 4-digit number that does not start with 0
- * and does not contain any duplicate digits.
- *
- * @return {number} A random 4-digit number
- */
+/* eslint-disable no-console */
+/* eslint-disable padding-line-between-statements */
 function generateRandomNumber() {
-  /* Write your code here */
+  let randomNumber;
+
+  do {
+    randomNumber = Math.floor(Math.random() * 9000) + 1000;
+  } while (new Set(randomNumber.toString()).size !== 4);
+
+  // eslint-disable-next-line no-console
+  console.log('Generated Random Number:', randomNumber);
```

Comment:
The console log statement here is used for debugging purposes. If this is not needed in the final version, consider removing or commenting it out to avoid unnecessary console output.